apps/arm/main.cpp

```diff
@@ -180,7 +181,85 @@ static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-    return MBED_SUCCESS; // TODO
+    /** ---------------------------------------------------------
+     * | 24 bits  | 32 bits  | 3 bits   | 1 bit   |  4 bits      |
+     * | unused   | data     | param    | vel/pos |  Actuator ID |
+     *  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    */
+    // Prepare bitfield to break down CAN data
+    struct{
+        // ID referring to which Actuator should be changed
+        // Param is Kp, Ki, Kd
+        unsigned int actuatorID : 4, param : 3;
+        // Bool for specifying velocity or position pid
+        bool velocity : 1;
+        // Actual value to be set. Bitfields can't have floats, use union
+        int32_t value : 32;
+    } payload;
+    union {
+        float float_value;
+        int32_t int_value;
+    }int_to_float;
```

Comment:
i don't like the definition of a union here. i think u should create a function to do this conversion. should maybe also call the union something like data. It feels wrong to call a union `int_to_float` because its not really something thats doing the conversion, its simply a data structure (sort of). 

If it is generecized and used elsewhere in the code base, we may want to abstract this into some conversions library for code reuse.